workflow/controller/scope.go

```diff
@@ -76,5 +78,30 @@ func (s *wfScope) resolveArtifact(v string) (*wfv1.Artifact, error) {
 	if !ok {
 		return nil, errors.Errorf(errors.CodeBadRequest, "Variable {{%s}} is not an artifact", v)
 	}
+
+	if subPath != "" {
+		// Copy resolved artifact pointer before adding subpath
+		copyArt := valArt.DeepCopy()
+
+		if copyArt.S3 != nil {
```

Comment:
there are 5 common artifact repos, I see only three branches?

Reply:
Yep, my bad. Didn't realise HDFS and OSS were also an option. Fixed it below.